client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -583,6 +585,22 @@ static Request clusterPutSettings(ClusterUpdateSettingsRequest clusterUpdateSett
         return new Request(HttpPut.METHOD_NAME, "/_cluster/settings", parameters.getParams(), entity);
     }
 
+    static Request clusterHealth(ClusterHealthRequest healthRequest) {
+        Params params = Params.builder();
+        params.withWaitForStatus(healthRequest.waitForStatus());
+        params.withWaitForNoRelocatingShards(healthRequest.waitForNoRelocatingShards());
+        params.withWaitForNoInitializingShards(healthRequest.waitForNoInitializingShards());
+        params.withWaitForActiveShards(healthRequest.waitForActiveShards());
+        params.withWaitForNodes(healthRequest.waitForNodes());
+        params.withTimeout(healthRequest.timeout());
+        params.withMasterTimeout(healthRequest.masterNodeTimeout());
+        params.withLocal(healthRequest.local());
+        params.putParam("level", "shards");
```

Comment:
I see. Right now you request the whole thing because it is simpler that way. I think this is fine. I think if we decide we'd like to support `level` as something else then we can do it in a followup. What do you think, @javanna?

Reply:
I had missed this line, so my comment on this is a bit off, sorry about that :) this is actually perfectly backwards compatible with the transport client behaviour, as it gets exactly everything. I think that we could add the level parameter though so that we can properly filter what gets returned at REST like every client works.